R/plot.R

```diff
@@ -2,62 +2,145 @@
 ### This file contains graphing functions for plotting the output from the ML_functions
 #############
 
-plot_scatter <- function(kfold_results) {
-  ggplot2::ggplot(kfold_results, ggplot2::aes(x=true, y=predicted)) +
+check_ids_match <- function(dfs) {
+  ids <- lapply(dfs, function(df) df$id)
+  all_equal <- lapply(ids, function(id_list) id_list == ids[[1]])
+  all(unlist(all_equal))
+}
+
+check_trues_match <- function(dfs) {
+  trues <- lapply(dfs, function(df) df$true)
+  all_equal <- lapply(trues, function(trues_list) all.equal(trues_list, trues[[1]]))
+  all(unlist(all_equal))
+}
+
+join_dfs <- function(dfs, keep_fold=FALSE) {
+  if(length(dfs) != 1) {
+    stopifnot(check_ids_match(dfs))
+    stopifnot(check_trues_match(dfs))
```

Comment:
Yes, that's exactly right. This was a clumsy way to get around the fact that I couldn't include ids in the model matrix. There are definitely safer ways to deal with this but this was quick and came to mind. I think it's fine for now but I'll add an issue to take another look at it when time permits. Good catch.